odoo/addons/base/models/res_users.py

```diff
@@ -919,15 +905,27 @@ def _is_superuser(self):
     def get_company_currency_id(self):
         return self.env.company.currency_id.id
 
+    @tools.ormcache()
     def _crypt_context(self):
         """ Passlib CryptContext instance used to encrypt and verify
         passwords. Can be overridden if technical, legal or political matters
         require different kdfs than the provided default.
 
-        Requires a CryptContext as deprecation and upgrade notices are used
-        internally
+        The work factor of the default KDF can be configured using the
+        ``password.hashing.rounds`` ICP.
         """
-        return DEFAULT_CRYPT_CONTEXT
+        cfg = self.env['ir.config_parameter'].sudo()
+        return CryptContext(
+            # kdf which can be verified by the context. The default encryption
+            # kdf is the first of the list
+            # TODO: provide key to *set* the alg?
+            ['pbkdf2_sha512', 'plaintext'],
+            # deprecated algorithms are still verified as usual, but
+            # ``needs_update`` will indicate that the stored hash should be
+            # replaced by a more recent algorithm.
+            deprecated=['auto'],
+            pbkdf2_sha512__rounds=int(cfg.get_param('password.hashing.rounds', 350000)),
```

Comment:
should we ignore the ICP if it's set below the default number of rounds? some people might think that 999 is a huge number of rounds, for example :thinking: 
(obviously they shouldn't be changing the ICP if they don't know what they're doing, but you know people...)

Reply:
Something like that?

Reply:
Could default to 0 now that I think about it, no need to bother repeating `MIN_ROUNDS`